### DISCUSSION TITLE
Bluetooth speaker connected into VS1053b

### DISCUSSION THREAD

**Discussion by SmokeyCheetah** [pschatzmann/ESP32-A2DP#484] [Q&A] [answered]
How to make to play audio via VS1053b (VS1053.h library) to it's speaker and make it Bluetooth speaker? That speaker connected do VS1053b?

This is my code
[main.zip](https://github.com/pschatzmann/ESP32-A2DP/files/12820033/main.zip)

Problem is this... something like read isn't there?

 size_t len = a2dp.**read**(mp3buff, sizeof(mp3buff));

if (len > 0) {
    player.playChunk(mp3buff, len);
  }

**Comment by TontonBidule** [comment]
I am also working on this problem, I tell you if I have anything new that can help :)

**Reply by SmokeyCheetah** [reply]
I have it on my work in school, so I really work hard on doing it 💯

**Comment by pschatzmann** [comment, marked as answer]
I started to work on this topic, but never had a chance to complete and test it. This [example](https://github.com/pschatzmann/arduino-audio-tools/blob/main/examples/sandbox/basic-a2dp-vs1053/basic-a2dp-vs1053.ino) that it in the sandbox of the audio tools might help you get started...

You might need to split up the out.write() into smaller pieces, because it is my gut feeling that the provided array might be too big.
